solidity_contracts/src/CairoPrecompiles/CairoCounterCaller.sol

```diff
@@ -32,11 +32,14 @@ contract CairoCounterCaller {
     }
 
     /// @notice Calls the Cairo contract to increment its internal counter
+    /// @dev The delegatecall preserves the caller's context, so the caller's address will
+    /// be the caller of this function.
     function incrementCairoCounter() external {
-        cairoCounter.callContract("inc");
+        cairoCounter.delegatecallContract("inc");
```

Comment:
With this delegatecall, now the storage of this function can be modified. 
The cairoCounter should be transformed as an `immutable` as a best practice.

Reply:
Is this actually a risk, knowing that we're actually calling `delegateCall` on a "precompile" address that will never touch storage? 

I'm open to making this immutable of course

Reply:
also `{call | staticcall | delegatecall}contract` is probably a bad naming because it doesn't explicitly indicate that we're calling a cairo contract.

I propose renaming to `{call | staticcall | delegatecall}Cairo`

Reply:
This is not a risk in this context as, as you said, this "precompile" will never touch storage.
And futur precompiles should not also.
I agree the naming `{call | staticcall | delegatecall}Cairo` is better

Note: my remark  above was more of a nitpick and best practice around delegatecall.
It is also a gas opti as access to immutable will be less expensive than a storage read.